srml/treasury/src/lib.rs

```diff
@@ -14,7 +14,48 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
-//! The Treasury: Keeps account of the taxed cash and handles its deployment.
+//! # Treasury Module
+//! 
+//! The `treasury` module keeps account of currency in a `pot` and manages the subsequent
+//! deployment of these funds.
+//! 
+//! ## Overview
+//! 
+//! Funds for treasury are raised in two ways:
+//! 1. By minting new tokens, leading to inflation, and
+//! 2. By channelling some of the tokens set for burning from transaction fees and slashing.
+//! 
+//! Treasury funds can be used to pay for developers who provide software updates,
+//! any changes decided by referenda, and to generally keep the system running smoothly. 
+//! 
+//! Treasury can be used with other modules, such as to tax validator rewards in the `staking` module.
+//! 
+//! ### Implementations 
+//! 
+//! The treasury module provides an implementation for the following trait:
+//! - `OnDilution` - Mint extra funds upon dilution; maintain the ratio of `portion` diluted to `total_issuance`.
```

Comment:
I don't think this should mention specific `variables`, especially ones that are generally handled outside this module. One can infer total issuance, but `portion` is a variable passed into this function and isn't defined anywhere here. Can you explain this trait in plain English?